thirdeye-ui/src/app/components/cancel-api-calls-on-page-unload/cancel-api-calls-on-page-unload.component.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import axios from "axios";
+import React, { FunctionComponent, useEffect, useState } from "react";
+
+/**
+ * This has to be wrapped around a page and not a router
+ */
+export const CancelAPICallsOnPageUnload: FunctionComponent = ({ children }) => {
+    const [controller] = useState(new AbortController());
+    axios.defaults.signal = controller.signal;
+
+    useEffect(() => {
+        axios.defaults.signal = controller.signal;
```

Comment:
As per the `axios.defaults` documentation, this should be applied globally (instead for just the children components), so would having this wrapper instantiated across multiple routes be any different from just being a wrapper around the root component??

Reply:
Yes this is different than having a wrapper at the root component. Having it at the root level will tag any api calls with the same signal across the app.

With this solution, developers have to explicitly wrap the pages they wish to cancel API calls on the page unload with the react router route definitions. This introduces some scoping as some API will be using whatever the abort signal is set to and that's unique and set by this component when its code is run.